File: mycollab-services/src/main/java/com/esofthead/mycollab/module/crm/dao/LeadMapperExt.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.crm.dao;

import com.esofthead.mycollab.core.persistence.IMassUpdateDAO;
import com.esofthead.mycollab.core.persistence.ISearchableDAO;
import com.esofthead.mycollab.module.crm.domain.Lead;
import com.esofthead.mycollab.module.crm.domain.SimpleLead;
import com.esofthead.mycollab.module.crm.domain.criteria.LeadSearchCriteria;
import org.apache.ibatis.annotations.Param;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public interface LeadMapperExt extends ISearchableDAO<LeadSearchCriteria>, IMassUpdateDAO<Lead, LeadSearchCriteria> {

    SimpleLead findById(Integer leadId);

    SimpleLead findConvertedLeadOfAccount(@Param("accountId") Integer accountId);

    SimpleLead findConvertedLeadOfContact(@Param("contactId") Integer contactId);

    SimpleLead findConvertedLeadOfOpportunity(@Param("opportunityId") Integer opportunity);
}

File: mycollab-services/src/main/java/com/mycollab/module/crm/dao/LeadMapperExt.kt
package com.mycollab.module.crm.dao

import com.mycollab.db.persistence.IMassUpdateDAO
import com.mycollab.db.persistence.ISearchableDAO
import com.mycollab.module.crm.domain.Lead
import com.mycollab.module.crm.domain.SimpleLead
import com.mycollab.module.crm.domain.criteria.LeadSearchCriteria
import org.apache.ibatis.annotations.Param

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
interface LeadMapperExt : ISearchableDAO<LeadSearchCriteria>, IMassUpdateDAO<Lead, LeadSearchCriteria> {

    fun findById(leadId: Int?): SimpleLead

    fun findConvertedLeadOfAccount(@Param("accountId") accountId: Int?): SimpleLead

    fun findConvertedLeadOfContact(@Param("contactId") contactId: Int?): SimpleLead

    fun findConvertedLeadOfOpportunity(@Param("opportunityId") opportunity: Int?): SimpleLead
}
